src/js/shim.js

```diff
@@ -0,0 +1,20 @@
+/* Copyright 2015-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+Number.isInteger = native.NumberIsInteger;
+Number.isSafeInteger = native.NumberIsSafeInteger;
+Number.isFinite = native.NumberIsFinite;
+Number.isNaN = native.NumberIsNaN;
```

Comment:
This needs to be moved to another PR, and BTW because ShadowNode includes the JerryScript, so shim is not necessary.